# HG changeset patch
# User icp <pangolin@vivaldi.net>
# Date 1673686673 -19800
#      Sat Jan 14 14:27:53 2023 +0530
# Branch 1.0.x
# Node ID cea9a8d772ba47b1cc3c9f0fc601f0e664c55797
# Parent  2b5d1f8787aa0dfa5c76240a2a53bc29f36c6166
tests: replace egrep calls to satisfy grep-v3.8+

diff --git a/tests/test-conflict-1.t b/tests/test-conflict-1.t
--- a/tests/test-conflict-1.t
+++ b/tests/test-conflict-1.t
@@ -16,13 +16,13 @@
   $ hg ci -m "A->C"
   created new head
 
-  $ hg merge -r1 2>&1 | sed 's/-C ./-C/' | egrep -v '^merging afile' | sed 's/incomplete.*/failed!/'
+  $ hg merge -r1 2>&1 | sed 's/-C ./-C/' | grep -E -v '^merging afile' | sed 's/incomplete.*/failed!/'
   warning: conflicts.* (re)
   0 files updated, 0 files merged, 0 files removed, 1 files unresolved
   use 'hg resolve' to retry unresolved file merges or 'hg *' to abandon (glob)
 resolve using first parent
   $ echo C > afile
-  $ hg resolve -m afile | egrep -v 'no more unresolved files' || true
+  $ hg resolve -m afile | grep -E -v 'no more unresolved files' || true
   $ hg ci -m "merge to C"
 
   $ hg log --graph --style compact | sed 's/\[.*\]//g'
diff --git a/tests/test-conflict-2.t b/tests/test-conflict-2.t
--- a/tests/test-conflict-2.t
+++ b/tests/test-conflict-2.t
@@ -16,13 +16,13 @@
   $ hg ci -m "A->C"
   created new head
 
-  $ hg merge -r1 2>&1 | sed 's/-C ./-C/' | egrep -v '^merging afile' | sed 's/incomplete.*/failed!/'
+  $ hg merge -r1 2>&1 | sed 's/-C ./-C/' | grep -E -v '^merging afile' | sed 's/incomplete.*/failed!/'
   warning: conflicts.* (re)
   0 files updated, 0 files merged, 0 files removed, 1 files unresolved
   use 'hg resolve' to retry unresolved file merges or 'hg *' to abandon (glob)
 resolve using second parent
   $ echo B > afile
-  $ hg resolve -m afile | egrep -v 'no more unresolved files' || true
+  $ hg resolve -m afile | grep -E -v 'no more unresolved files' || true
   $ hg ci -m "merge to B"
 
   $ hg log --graph --style compact | sed 's/\[.*\]//g'
diff --git a/tests/test-git-submodules.t b/tests/test-git-submodules.t
--- a/tests/test-git-submodules.t
+++ b/tests/test-git-submodules.t
@@ -27,10 +27,10 @@
 while older git will use the full normalized path for .)
   $ clonefilt='s/Cloning into/Initialized empty Git repository in/;s/in .*/in .../'
 
-  $ git clone ../gitrepo1 . 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | egrep -v '^done\.$'
+  $ git clone ../gitrepo1 . 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | grep -E -v '^done\.$'
   Initialized empty Git repository in ...
   
-  $ git submodule add ../gitsubrepo subrepo 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | egrep -v '^done\.$'
+  $ git submodule add ../gitsubrepo subrepo 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | grep -E -v '^done\.$'
   Initialized empty Git repository in ...
   
   $ git commit -m 'add subrepo' | sed 's/, 0 deletions(-)//'
@@ -48,7 +48,7 @@
   [master a000567] change subrepo commit
    1 file changed, 1 insertion(+), 1 deletion(-)
 
-  $ git submodule add ../gitsubrepo subrepo2 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | egrep -v '^done\.$'
+  $ git submodule add ../gitsubrepo subrepo2 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | grep -E -v '^done\.$'
   Initialized empty Git repository in ...
   
   $ git commit -m 'add another subrepo' | sed 's/, 0 deletions(-)//'
@@ -79,7 +79,7 @@
 
   $ git rm alpha
   rm 'alpha'
-  $ git submodule add ../gitsubrepo alpha 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | egrep -v '^done\.$'
+  $ git submodule add ../gitsubrepo alpha 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | grep -E -v '^done\.$'
   Initialized empty Git repository in ...
   
   $ git commit -m 'replace file with subrepo' | sed 's/, 0 deletions(-)//' | sed 's/, 0 insertions(+)//'
@@ -98,7 +98,7 @@
 
   $ git rm foolink
   rm 'foolink'
-  $ git submodule add ../gitsubrepo foolink 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | egrep -v '^done\.$'
+  $ git submodule add ../gitsubrepo foolink 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | grep -E -v '^done\.$'
   Initialized empty Git repository in ...
   
   $ git commit -m 'replace symlink with subrepo' | sed 's/, 0 deletions(-)//' | sed 's/, 0 insertions(+)//'
diff --git a/tests/test-renames.t b/tests/test-renames.t
--- a/tests/test-renames.t
+++ b/tests/test-renames.t
@@ -39,7 +39,7 @@
   $ rmpwd="import sys; print(sys.stdin.read().replace('$(dirname $(pwd))/', ''))"
   $ clonefilt='s/Cloning into/Initialized empty Git repository in/;s/in .*/in .../'
 
-  $ git submodule add ../gitsubmodule 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | egrep -v '^done\.$'
+  $ git submodule add ../gitsubmodule 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | grep -E -v '^done\.$'
   Initialized empty Git repository in ...
   
   $ fn_git_commit -m 'add submodule'
@@ -72,7 +72,7 @@
 Rename a file elsewhere and replace it with a submodule:
 
   $ git mv gamma gamma-new
-  $ git submodule add ../gitsubmodule gamma 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | egrep -v '^done\.$'
+  $ git submodule add ../gitsubmodule gamma 2>&1 | python -c "$rmpwd" | sed "$clonefilt" | grep -E -v '^done\.$'
   Initialized empty Git repository in ...
   
   $ fn_git_commit -m 'rename and add submodule'
diff --git a/tests/test-subrepos.t b/tests/test-subrepos.t
--- a/tests/test-subrepos.t
+++ b/tests/test-subrepos.t
@@ -28,7 +28,7 @@ allow commits despite working copy prese
   $ git config receive.denyCurrentBranch ignore
   $ cd ..
 Ensure gitlinks are transformed to .hgsubstate on hg pull from git
-  $ hg clone -u tip gitrepo1 hgrepo 2>&1 | egrep -v '^(Cloning into|done)'
+  $ hg clone -u tip gitrepo1 hgrepo 2>&1 | grep -E -v '^(Cloning into|done)'
   importing git objects into hg
   updating to branch default
   cloning subrepo subrepo1 from $TESTTMP/gitsubrepo
--- hg-git-0.10.4/tests/testutil
+++ hg-git-0.10.4/tests/testutil
@@ -22,7 +22,7 @@ echo '[subrepos]' >> $HGRCPATH
 echo 'git:allowed = yes' >> $HGRCPATH
 
 # Disable warnings on 4.6, specifically; clone triggers a warning
-if hg --version | fgrep -q 'version 4.6'
+if hg --version | grep -F -q 'version 4.6'
 then
     echo "[devel]" >> $HGRCPATH
     echo "all-warnings=no" >> $HGRCPATH
